### ISSUE TITLE
@babel/core parseFile

### ISSUE THREAD

**Issue by Gallevy** [babel/babel#11104]
## Feature Request

- [ ] I would like to work on this feature!

**Is your feature request related to a problem? Please describe.**
ParseFile method to parse code from a file into AST

**Describe the solution you'd like**
a parseFile method, similar to transformFile but just to parse into AST

**Describe alternatives you've considered**
using @babel/core transformFile and using the result AST

**Teachability, Documentation, Adoption, Migration Strategy**
Same as parse but to parse a whole file

**Comment by kaicataldo** [comment]
Mind adding some more information about why the existing [`parseSync`](https://babeljs.io/docs/en/babel-core#parsesync) and [`parseAsync`](https://babeljs.io/docs/en/babel-core#parseasync) methods don't cover your use case?

**Comment by Gallevy** [comment]
I want to read an entire directory. Methods like transformFile do help you read entire files. But with parse you don't have parseFile which leaves me with two options. Either read the file myself and use the parse method with the read file or use transformFile and only care about the AST

**Comment by kaicataldo** [comment]
> I want to read an entire directory.

Just to be clear, are you expecting the `parseFile` method to do this? That seems out of scope.

I'd like to hear from other team members, but I personally don't think this proposal meets the bar for inclusion in core, since you would need to running this programmatically anyway and iterating through a directory of files, reading their contents, and calling the parse method on these contents isn't a whole lot of overhead, while maintaining yet another public API is.

**Comment by Gallevy** [comment]
> > I want to read an entire directory.
> 
> Just to be clear, are you expecting the `parseFile` method to do this? That seems out of scope to me.
No, just read a single file, similar to how transformFile works
